tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/ChunkMetadata.java

```diff
@@ -198,30 +197,35 @@ public void setVersion(long version) {
   }
 
   public List<TimeRange> getDeleteIntervalList() {
-    return deleteIntervalSet == null ? null : new ArrayList<>(deleteIntervalSet);
+    return deleteIntervalList;
   }
 
-  public void insertIntoSortedDeletions(long startTime, long endTime) {
-    TimeRange timeRange = new TimeRange(startTime, endTime);
-
-    if (deleteIntervalSet != null) {
-      TimeRange interval = deleteIntervalSet.floor(timeRange);
-      while (interval != null && timeRange.intersects(interval)) {
-        timeRange.merge(interval);
-        deleteIntervalSet.remove(interval);
-        interval = deleteIntervalSet.floor(timeRange);
+  @Override
+  public void insertIntoSortedDeletions(TimeRange timeRange) {
+    List<TimeRange> resultInterval = new ArrayList<>();
+    long startTime = timeRange.getMin();
+    long endTime = timeRange.getMax();
+    if (deleteIntervalList != null) {
+      if (deleteIntervalList.get(deleteIntervalList.size() - 1).getMax() < timeRange.getMin()) {
+        deleteIntervalList.add(timeRange);
+        return;
       }
-      interval = deleteIntervalSet.ceiling(timeRange);
-      while (interval != null && timeRange.intersects(interval)) {
-        timeRange.merge(interval);
-        deleteIntervalSet.remove(interval);
-        interval = deleteIntervalSet.ceiling(timeRange);
+      for (TimeRange interval : deleteIntervalList) {
+        if (interval.getMax() < startTime) {
+          resultInterval.add(interval);
+        } else if (interval.getMin() > endTime) {
+          resultInterval.add(new TimeRange(startTime, endTime));
+          startTime = interval.getMin();
+          endTime = interval.getMax();
```

Comment:
It seems that you can add all the remaining `TimeRange` in `deleteIntervalList` and then break the loop early here, because the remaining intervals won't overlap.

Reply:
done